.github/workflows/pre-release.yml

```diff
@@ -0,0 +1,168 @@
+          else
+            echo "Creating new release branch release-$MAJOR_MINOR_VERSION"
+            git checkout -b release-$MAJOR_MINOR_VERSION
+            git push origin release-$MAJOR_MINOR_VERSION
+          fi          
+          git checkout release-prep-${{ inputs.new_version }}
+      - name: Set up the Go@${{ env.GO_VERSION }} environment
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Update the VERSION manifest
+        run: echo "${{ inputs.new_version }}" | sed 's/^v//' > VERSION
+      - name: update data.yaml
+        run: |
+            chmod +x scripts/update-data-yaml.sh
+            ./scripts/update-data-yaml.sh "${{ inputs.new_version }}"
+      - name: Update the compatibility matrix (README.md)
+        run: make generate
+      - name: Generate the release date for unreleased (CHANGELOG.md)
+        run: |
+          chmod +x scripts/generate-release-date.sh
+          ./scripts/generate-release-date.sh "${{ inputs.new_version }}"
+      - name: Lint
+        run: |
+          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | \
+          sh -s -- -b $(go env GOPATH)/bin ${{ env.GOLANGCI_LINT_VERSION }}
+          make lint-fix
+      - name: Generate manifests
+        run: |
+            make clean || true
+            VERSION="${{ inputs.new_version }}" make examples
+      - name: Run rule tests
+        run: PROMTOOL_CLI=./promtool make install-promtool test-rules
+      - name: Run unit tests
+        run: make test-unit
+      - name: Run end-to-end tests
+        run: |
+          make e2e
+          find examples -name "*.yaml" -type f -exec sed -i 's|kube-state-metrics-amd64|kube-state-metrics|g; s|kube-state-metrics-arm64|kube-state-metrics|g' {} \;
+      - name: Update the remote and commit the changes
+        run: |
+          git config --local user.email "ksm-release-bot@mock-k8s.io"
+          git config --local user.name "KSM Release Bot"
+          git add .
+          git commit -m "chore: Cut ${{ inputs.new_version }}"
+          MAJOR_MINOR_VERSION=$(echo "${{ inputs.new_version }}" | sed 's/^v//' | cut -d. -f1,2)
+          git push origin release-prep-${{ inputs.new_version }}
+      - name: 'Run benchmark tests'
+        run: |
+          BENCHSTAT_OUTPUT_FILE=result.txt ./tests/compare_benchmarks.sh main 2
+      - name: Post results to job summary
+        run: |
+          echo "### Benchmark Comparison Results" >> "$GITHUB_STEP_SUMMARY"
+          echo "\`\`\`" >> "$GITHUB_STEP_SUMMARY"
+          cat result.txt >> "$GITHUB_STEP_SUMMARY"
+          echo "\`\`\`" >> "$GITHUB_STEP_SUMMARY"
+      - name: Validate docs
+        run: make doccheck
+      - name: Validate manifests
+        run: make validate-manifests
+      - name: Validate go modules
+        run: make validate-modules
+      - name: Create a pull request
```

Comment:
I'd appreciate if the longer lines could be broken (using multi-line separators) down and commented for this step.